include/ccf/pal/snp_ioctl6.h

```diff
@@ -140,4 +198,50 @@ namespace ccf::pal::snp::ioctl6
       return {quote_bytes, quote_bytes + resp_wrapper.resp.report_size};
     }
   };
-}
+
+  class DerivedKey
+  {
+    DerivedKeyRespWrapper resp_wrapper = {};
+
+  public:
+    DerivedKey()
+    {
+      int fd = open(DEVICE, O_RDWR | O_CLOEXEC);
+      if (fd < 0)
+      {
+        throw std::logic_error(
+          fmt::format("Failed to open \"{}\" ({})", DEVICE, fd));
+      }
+
+      // This req by default mixes in HostData and the CPU VCEK
+      DerivedKeyReq req = {};
+      // We must also mix in the measurement
+      req.guest_field_select.measurement = 1;
+      GuestRequestDerivedKey payload = {
+        .req_data = &req, .resp_wrapper = &resp_wrapper, .exit_info = {0}};
+      int rc = ioctl(fd, SEV_SNP_GUEST_MSG_DERIVED_KEY, &payload);
+      if (rc < 0)
+      {
+        LOG_FAIL_FMT("IOCTL call failed: {}", strerror(errno));
+        LOG_FAIL_FMT(
+          "Exit info, fw_error: {} vmm_error: {}",
+          payload.exit_info.errors.fw,
+          payload.exit_info.errors.vmm);
+        throw std::logic_error(
+          "Failed to issue ioctl SEV_SNP_GUEST_MSG_DERIVED_KEY");
+      }
+      if ((*payload.resp_wrapper).resp.status != 0)
+      {
+        LOG_FAIL_FMT(
+          "SNP_GUEST_DERIVED_KEY failed: {}", resp_wrapper.resp.status);
+        throw std::logic_error(
+          "Failed to issue ioctl SEV_SNP_GUEST_MSG_DERIVED_KEY");
+      }
+    }
+
+    const std::span<const uint8_t> get_raw()
```

Comment:
```suggestion
    std::span<const uint8_t> get_raw()
```

I don't think `std::span` actually has any mutable methods? But if it had something like a `.shrink()` method we should be able to call it on this instance, without creating a further copy. Generally a bad pattern to return `const` values, unless it's some kind of handle/reference to something that's owned elsewhere.